src/alert/alert.ts

```diff
@@ -51,9 +53,10 @@ export class NgbAlert {
  * Alerts that can be dismissed without any additional code.
  */
 @Directive({selector: 'template[ngbAlert]'})
-export class NgbDismissibleAlert implements OnInit {
+export class NgbDismissibleAlert implements OnInit, OnDestroy {
   private _windowFactory: ComponentFactory<NgbAlert>;
   private _windowRef: ComponentRef<NgbAlert>;
+  private timeout: number;
```

Comment:
as today we are prefixing private with underscore.

I am going to change that probably but as today, you need to change it.